src/couch_httpd_auth.erl

```diff
@@ -304,12 +310,19 @@ handle_session_req(#httpd{method='POST', mochi_req=MochiReq}=Req, AuthModule) ->
             UserSalt = couch_util:get_value(<<"salt">>, UserProps2),
             CurrentTime = make_cookie_time(),
             Cookie = cookie_auth_cookie(Req, ?b2l(UserName), <<Secret/binary, UserSalt/binary>>, CurrentTime),
+            % maybe set the csrf token
+            Cookies = case couch_httpd:qs_value(Req, "csrf", "false") of
```

Comment:
mmm...so we here
1. Enable CSRF protection only by user request, not for the whole server. What means, that it will be easy to forget to activate that protection.
2. CSRF is not applied for Basic and OAuth authed users and those that authed using other methods as they don't POST /_session. They could only GET it, but it's not mandatory. 

What are your thoughts about these cases?